core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/TrainPhysicsIntegrator.java

```diff
@@ -74,7 +74,7 @@ private IntegrationStep step(
     private IntegrationStep step(double timeStep, double position, double speed) {
         double tractionForce = 0;
         double brakingForce = 0;
-        var tractiveEffortCurve = tractiveEffortCurveMap.get(position);
+        var tractiveEffortCurve = tractiveEffortCurveMap.get(Math.min(Math.max(0, position), path.getLength()));
```

Comment:
Isn't there a way we make sure `position` is in an acceptable range before this line ?

Reply:
Do you mean to ask if there is a way to avoid having to clamp the value through the algorithm itself, or do you mean clamping the value earlier in the function ?

Reply:
Well I was surprised too you see
It's necessary because of the positionDelta which leads to ever so slightly exit the path.
That's why it's applied in `getWeightForce` for example

Reply:
But how come the positionDelta leads to an exit of the path now ? I didn't have a look in depth in the rest of the PR but if this change is necessary that _might_ need something's wrong somewhere (i'm really not saying that's the case, i'm just wondering).

I agree with the change anyways, as it cannot lead to wrong results, but i'm still surprised

Reply:
In fact it always did before, it's just that before the mapping extended out of the path, and now not anymore

Reply:
Alright, that's clearer, thanks for investigating ! :)